ggshield/utils/files.py

```diff
@@ -43,7 +44,16 @@ def get_filepaths(
                 raise UnexpectedDirectoryError(path)
 
             if not ignore_git and is_git_dir(path):
-                _targets = {path / target for target in git_ls(path)}
+                if ignore_git_staged:
+                    _targets = {
+                        path / target
+                        for target in get_filepaths_from_ref(
+                            ref="HEAD",
+                            wd=path,
+                        )
+                    }
+                else:
+                    _targets = {path / target for target in git_ls(path)}
```

Comment:
nit: this can be factorized a bit. Something like this:

```python
paths = get_filepaths_from_ref("HEAD", wd=path) if ignore_git_staged else git_ls(path)
_targets = {path / x for x in paths}
```